engine/overworld/weather.asm

```diff
@@ -0,0 +1,778 @@
+DoOverworldWeather:
+	push de
+	push hl
+	push bc
+	ld hl, wWeatherFlags
+	bit OW_WEATHER_DISABLED_F, [hl]
+	jr nz, .done
+	ld a, [wOverworldWeatherDelay]
+	and %1 ; 30 fps
+	jr z, .done
+	ld a, [wOverworldWeatherCooldown]
```

Comment:
I would love more comments on this file; from simple ones like "`; decrement weather cooldown, capped at 0`" on this "paragraph", to a broader explanation of what functions are for, or how the system interacts. I'm not saying it needs an entire document explaining such things, just, more context clues in long runs of code like this.

Reply:
Applied more comments throughout; let me know if you need more of a specific kind.